### ISSUE TITLE
Selecting the best machine learning pipeline

### ISSUE THREAD

**Issue by RLeenings** [ohbm/hackathon2023#8]
### Title

PHOTONAI

## INTRODUCTION
With photonai, we have designed a machine learning software that abstracts and condenses machine learning analyses to the most important design decisions, so that it can be applied from practitioners in medicine and the Life Sciences. As a part of that, we have automated the nested cross-validation and hyperparameter optimization loop. However, the optimal approach for choosing the most suitable hyperparameter configuration continues to be an open problem. For example, the hyperparameter configuration that performs best on the validation set might be overly specific to the particularities of the validation set and thus underperform in new applications. 

## GOALS FOR THE BRAIN HACK

- brainstorm strategies on how to select the best hyperparameter configuration to maximize generalization performance
- implement ensembling techniques, i.e. select several hyperparameter configurations 

### Link to the Project

github.com/wwu-mmll/photonai

### Project lead

Jan Ernsting, Github: jernsting, Discord: Jan#5471
Ramona Leenings, Github: rleenings, Discord: ramona_photonai

### Main Hub

Montreal

### Other Hub covered by the leaders

- [ ] Montreal
- [ ] Asia / Pacific
- [ ] Europe / Middle East / Africa
- [ ] Americas

### Skills

Basic Python
Basic Machine Learning or Statistic Knowledge

### Recommended tutorials for new contributors

- [X] [Set up a brainhack friendly computing environment](https://school-brainhack.github.io/modules/installation/)
- [ ] [A introduction to Bash](https://school-brainhack.github.io/modules/introduction_to_terminal/)
- [X] [Python: Writing a script](https://school-brainhack.github.io/modules/python_scripts/)
- [X] [Python: Data analysis with Python](https://school-brainhack.github.io/modules/python_data_analysis/)
- [ ] [Python: Visualisation](https://school-brainhack.github.io/modules/python_visualization/)
- [X] [Machine learning basics](https://school-brainhack.github.io/modules/machine_learning_basics/)
- [ ] [BIDS](https://school-brainhack.github.io/modules/bids/)
- [ ] [Machine learning for neuroimaging](https://school-brainhack.github.io/modules/machine_learning_neuroimaging/)
- [X] [VCS: Using Git and Github](https://school-brainhack.github.io/modules/git_github/)
- [ ] [VCS: Data management with Datalad](https://school-brainhack.github.io/modules/datalad/)
- [ ] [Virtualization](https://school-brainhack.github.io/modules/containers/)
- [ ] High Performance Computing: [Oracle cloud resource](https://brainhack.org/brainhack_cloud/)

### Short name for the Discord chat channel (~15 chars)

photonai

### Please read and follow the OHBM Code of Conduct

- [X] I agree to follow the OHBM Code of Conduct during the hackathon

**Comment by ufangYang** [comment]
hi @bhvieira  can you provide the content and/or figure for this project to be added to Brainhack Proceeding 2023 by 15th May? Also, please don't forget to fill out your team member's details using this form: https://forms.gle/Ku3AVHcXvpUqVVkV7 . Cheers
